scripts/ci-e2e.sh

```diff
@@ -89,8 +89,15 @@ source "${REPO_ROOT}/hack/ensure-kustomize.sh"
 
 # This will run the tests with env variabls defined in environment.sh
 # or exported by metal3-dev-env scripts
-${M3_DEV_ENV_PATH}/scripts/run_command.sh make e2e-tests
+${M3_DEV_ENV_PATH}/scripts/run_command.sh make e2e-tests || sleep infinity
 
 pushd ${M3_DEV_ENV_PATH} || exit 1
 make clean
 popd || exit 1
+
+if upgradeTest then
+  docker pull quay.io/metal3-io/cluster-api-provider-metal3:release-0.5
+  docker image tag quay.io/metal3-io/cluster-api-provider-metal3:release-0.5 192.168.111.1:5000/localimages/cluster-api-provider-metal3:release-0.5
+  docker pull quay.io/metal3-io/ip-address-manager:release-0.1
+  docker image tag quay.io/metal3-io/ip-address-manager:release-0.1 192.168.111.1:5000/localimages/ip-address-manager:release-0.1
+fi
```

Comment:
shouldn't this happen before the `${M3_DEV_ENV_PATH}/scripts/run_command.sh make e2e-tests` command?